File: baseProjectPY/pipelines/basePipeline1/code/basepipeline1/udfs/UDFs.py
from pyspark.sql import *
from pyspark.sql.functions import *
from pyspark.sql.types import *
from prophecy.lookups import (
    createLookup,
    createRangeLookup,
    lookup,
    lookup_last,
    lookup_match,
    lookup_count,
    lookup_row,
    lookup_row_reverse,
    lookup_nth
)
y = 10

def registerUDFs(spark: SparkSession):
    spark.udf.register("udfConcat", udfConcat)
    spark.udf.register("udfConcatold", udfConcatold)

def udfConcatGenerator():
    delimitter = ' '

    @udf(returnType = StringType())
    def func(value: str, value2: str):
        return value + delimitter + value2

    return func

udfConcat = udfConcatGenerator()

@udf(returnType = StringType())
def udfConcatold(value: str, value2: str):
    return value + value2

File: finalMigration/pipelines/p2/code/src/main/scala/io/prophecy/pipelines/p1/udfs/UDFs.scala
package io.prophecy.pipelines.p1.udfs

import _root_.io.prophecy.abinitio.ScalaFunctions._
import _root_.io.prophecy.libs._
import org.apache.spark.sql.types._
import org.apache.spark.sql.functions._
import org.apache.spark.sql._

object UDFs extends Serializable {
  val x = 1

  def registerUDFs(spark: SparkSession) = {
    spark.udf.register("udf1", udf1)
    spark.udf.register("udf2", udf2)
    spark.udf.register("udf3", udf3)
  }

  def udf1 = udf(() => "1")
  def udf2 = udf(() => "1")
  def udf3 = udf(() => "1")
}
